src/util/expr_iterator.h

```diff
@@ -185,18 +184,11 @@ class depth_iterator_baset
     for(auto &state : m_stack)
     {
       // This deliberately breaks sharing as expr is now non-const
-      auto &operands = expr->operands();
-      // Get iterators into the operands of the new expr corresponding to the
-      // ones into the operands of the old expr
-      const auto it = operands.end() - (state.end - state.it);
+      (void)expr->operands();
```

Comment:
Maybe that should be `(void)expr->write();` instead of a somewhat random function (`operands`)?